### PR TITLE
Add gas schedule

### PR DESCRIPTION
**Pull request by jackzhhuang**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

-
-
-

### CODE REVIEW

**Comment by nkysg** [comment]
https://github.com/starcoinorg/starcoin-framework/pull/234 @jackzhhuang 和这个类似

File: vm/vm-runtime/src/starcoin_vm.rs [hunk 1]
```diff
@@ -99,6 +102,7 @@ impl StarcoinVM {
             move_version: None,
             native_params,
             gas_params: Some(gas_params),
+            gas_schedule: None,
```
Comment (nkysg) [inline vm/vm-runtime/src/starcoin_vm.rs:105]: 这 not(feature = "metrics")也需要处理

Comment (jackzhhuang) [reply vm/vm-runtime/src/starcoin_vm.rs:105]: 好的，我补上

File: vm/vm-runtime/src/starcoin_vm.rs [hunk 2]
```diff
@@ -121,26 +125,30 @@ impl StarcoinVM {
     pub fn load_configs<S: StateView>(&mut self, state: &S) -> Result<(), Error> {
         if state.is_genesis() {
             self.vm_config = Some(VMConfig {
-                gas_schedule: G_LATEST_GAS_SCHEDULE.clone(),
+                gas_schedule: G_LATEST_GAS_COST_TABLE.clone(),
             });
             self.version = Some(Version { major: 1 });
+            self.gas_schedule = Some(GasSchedule::from(
+                &self.vm_config.as_ref().unwrap().gas_schedule,
```
Comment (nkysg) [inline vm/vm-runtime/src/starcoin_vm.rs:133]: 这里是不是直接用G_LATEST_GAS_COST_TABLE可以，不用&self.vm_config.as_ref().unwrap().gas_schedule,

File: vm/vm-runtime/src/starcoin_vm.rs [hunk 3]
```diff
@@ -121,26 +125,30 @@ impl StarcoinVM {
...
-                    debug!("update native_params");
-                    Arc::get_mut(&mut self.move_vm)
-                        .unwrap()
-                        .update_native_functions(params.clone().natives)?;
-                    self.native_params = params.natives.clone();
-                }
-                self.gas_params = gas_params;
+        let gas_params = StarcoinGasParameters::from_on_chain_gas_schedule(
+            &self.gas_schedule.as_ref().unwrap().clone().to_btree_map(),
+        );
```
Comment (nkysg) [inline vm/vm-runtime/src/starcoin_vm.rs:143]: 这里尽量不用unwrap,以免直接panic

File: vm/vm-runtime/src/starcoin_vm.rs [hunk 4]
```diff
@@ -121,26 +125,30 @@ impl StarcoinVM {
...
+        if let Some(ref params) = gas_params {
+            if params.natives != self.native_params {
+                debug!("update native_params");
+                Arc::get_mut(&mut self.move_vm)
+                    .unwrap()
```
Comment (nkysg) [inline vm/vm-runtime/src/starcoin_vm.rs:148]: 这里是我之前写的，也应该看看能否直接报错，不用unwrap

File: vm/vm-runtime/src/starcoin_vm.rs [hunk 5]
```diff
@@ -222,15 +237,43 @@ impl StarcoinVM {
...
+                        .pop()
+                        .ok_or_else(|| {
+                            anyhow::anyhow!("Expect 0x1::GasSchedule::initialize() return value")
+                        })?;
+                    bcs_ext::from_bytes::<GasSchedule>(&data)?
+                };
+                (Some(gas_schedule), "gas schedule from gas schedule in move")
+            };
+            #[cfg(feature = "print_gas_info")]
+            self.gas_schedule.as_ref().unwrap().info(_message);
```
Comment (nkysg) [inline vm/vm-runtime/src/starcoin_vm.rs:277]: 这里也是

**Comment by nkysg** [comment]
the commit is in https://github.com/starcoinorg/starcoin/pull/3889

**Pull request closed without merging**